spec/unit/puppet/provider/conf_setting/ruby_spec.rb

```diff
@@ -110,6 +110,90 @@ def validate_file(expected_content,tmpfile = tmpfile)
       EOS
 )
     end
+
+    it "should add an array element even if the target setting does not yet exist" do
+      resource = Puppet::Type::Hocon_setting.new(
+          common_params.merge(:setting => 'test_key_2',
+                              :ensure => 'present',
+                              :value => { 'foo' => 'foovalue3' },
+                              :type => 'array_element'))
+      provider = described_class.new(resource)
+      expect(provider.exists?).to be false
+      provider.create
+      validate_file(
+          <<-EOS
+test_key_1: [
+  {
+    foo: foovalue
+    bar: barvalue
+    master: true
+  }
+,
+  {
+    foo: foovalue2
+    baz: bazvalue
+    url: "http://192.168.1.1:8080"
+  }
+,
+  {
+    foo: foovalue3
+  }
+]
+test_key_2 : [
+
+    {
+
+        "foo" : "foovalue3"
+
+    }
+
```

Comment:
@camlow325 That's correct, the whitespace behavior is controlled entirely by the hocon gem. It definitely seems weird to me that there seem to be some extra newlines being generated, so maybe there are some improvements that need to be made there.